std/datetime.d

```diff
@@ -1075,6 +1075,42 @@ public:
         //assert(ist.opCmp(ist) == 0);
     }
 
+    /**
+     * Returns: A hash of the $(LREF SysTime)
+     */
+    size_t toHash() const @nogc pure nothrow @safe
+    {
+        static if (is(size_t == ulong))
+        {
+            return _stdTime;
+        }
+        else
+        {
+            enum uint m = 0x5bd1e995;
+            enum uint n = 16;
+            enum uint r = 24;
+            uint h = n;
+            auto k = _stdTime;
```

Comment:
`toHash` is const, so auto is inferring `typeof(k) == const(long)`

Switch to `long k = _stdTime`